lib/clAdapters/google-mail/index.js

```diff
@@ -229,21 +271,18 @@ var convertEmailListToArrayOfEmailAddressObjects = function(emailList) {
 };
 
 var hasLabel = function(message, labelValue) {
-  return message.labelIds.indexOf(labelValue) >= 0;
+  return message.labelIds && message.labelIds.length && (message.labelIds.indexOf(labelValue) >= 0);
 };
 
 var mapEmailData = function(emailData) {
   var mappedData = [];
 
   for (var userIter = 0; userIter < emailData.length; userIter++) {
-    var mappedUser = {
-      email: emailData[userIter].email,
-      filterStartDate: emailData[userIter].filterStartDate,
-      filterEndDate: emailData[userIter].filterEndDate,
+    var mappedUser = _.assign({}, emailData[userIter], {
       data: [],
       success: emailData[userIter].success,
       errorMessage: emailData[userIter].errorMessage
-    };
+    });
 
     if (emailData[userIter].success) {
       for (var i = 0; i < emailData[userIter].data['messages'].length; i++) {
```

Comment:
Might want to cache the length prop here as this is several properties deep and in an inner loop so might get expensive